### PR TITLE
docs: mention passing the .tsx file to createPage

### PR DESCRIPTION
**Pull request by dandv**
Fix for #23328.

### CODE REVIEW

File: packages/gatsby-plugin-typescript/README.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ module.exports = {
 }
 ```
 
+When [creating pages programmatically](https://www.gatsbyjs.org/docs/programmatically-create-pages-from-data/#creating-pages), you can pass the `.tsx` filename directly as the `component` for [`createPage`](https://www.gatsbyjs.org/docs/actions/#createPage).
```
Comment (LekoArts) [inline packages/gatsby-plugin-typescript/README.md:24]:
```suggestion
When [creating pages programmatically](/docs/programmatically-create-pages-from-data/#creating-pages), you can pass the `.tsx` filename directly as the `component` for [`createPage`](/docs/actions/#createPage).
```

Please use relative links :)

**Comment by LekoArts** [comment]
So technically this is correct, however I personally feel like this is a bit repetitive / adds noise as we implicitly state that it works (as we don't explicitly state that it **doesn't** work). However, I have no strong opinions on adding it nevertheless so I'll leave it to the learning team to decide on whether it's worth adding it or not :)

File: packages/gatsby-plugin-typescript/README.md (hunk 1 above)
Comment (Ekwuno) [inline packages/gatsby-plugin-typescript/README.md:24]: Seeing as this is suggested but not explicitly stated, I would rather add this into the doc as a note. 
```suggestion
 > When [creating pages programmatically](/docs/programmatically-create-pages-from-data/#creating-pages), you can pass the `.tsx` filename directly as the `component` for [`createPage`](/docs/actions/#createPage).
```

**Comment by dandv** [comment]
The reason I left this PR is because I couldn't just use `.ts` everywhere. For example, `gatsby-node.ts` isn't parsed. It has to be `.js` or `.cjs`

**Pull request merged**